### DISCUSSION TITLE
What would you like to see on NetBox's main page?

### DISCUSSION THREAD

**Discussion by jeremystretch** [netbox-community/netbox#5500] [General]
Other than some minor additions, NetBox's main page hasn't changed much since its release over four years ago:

![NetBox main page](https://user-images.githubusercontent.com/13487278/102655029-9a282780-413f-11eb-9339-3c5066028276.png)

I've been absently pondering how we might spruce it up a bit, and I'm curious to hear what people would like to see there.

**Comment by nahun** [comment]
Widgets that users can add, remove, and move (and the dashboard setup be per user) would be nice. But that might be asking a lot.

I don't have many ideas, but here are a few:
- Allow plugins to show widgets or data
- Sorted prefix utilization list
- List of favorited objects by the user
- List of circuit providers
- Show the last run of a given report. Not just a link, but show the actual report results.

**Reply by adeeb1981** [reply]
highly recommended to display home as Widgets that users can add, remove, and arrange, also store the customized view in his preference

**Comment by dejantep** [comment]
I think change log could move somewhere else or disappear completely. An alternative to that could be to have "badge/alert" on respective web part showing if something change since last login

**Comment by xkilian** [comment]
- Filter change log using tags or regex, in user preferences, so that the change list on the main page is more pertinent to the user or drop it completely.
- Improving rôle based workflows in the main page. Say for example, I am user that mainly works with cabling, I would like to have a bunch of pages related to this workflow so that his work is more streamlined. So making it easier to streamlined various jobs. Of course, this may be a candidate for a custom plugin that takes in to account some busines logic and validation, but having general job related efficiencies in navigation is worth pondering IMO. This may be more of a thought for the next version of netbox of the netbox UI. The current UI, is a general overview of what is in the system.
- I guess the same could be said for sub-pages. If I have more facilities related rôle (power, heat, rack usage/device placement, weight(not currently supported by netbox). Then I might want some sort of customized view for my rôle. Versus an IP or cabling person.

**Comment by evatych** [comment]
Rack Groups. We very often need to watch a specific group.

**Comment by rodvand** [comment]
- The ability to hide certain sections (when we're not utilising the Secrets function of NetBox I don't need to see it on the front page).
- Ability to add links to other tools (this could also be moved to a plugin if the plugin can add data to the front page). When NetBox is the first place we want our engineers to go it would be nice to enable them to navigate to some of our other tools as well from NetBox.
- Allow users to customise the front page to their needs.

**Comment by chubblyButs** [comment]
Perhaps a widget with top used objects for example most used device type, or most used cable size. Tenant with most racks, devices, etc.
I like the cable size one, it would tell me what size cable I need to order more of.  Also maybe a widget for admins with the list of current connected users.

**Comment by lastwednesday** [comment]
Some ratios of infrastructure use could be helpful to track eg:
1) The total number of interfaces which are cabled vs the total number which are free.
2) The number of rack space open vs total rack space
3) Power in use vs available power

Though those might be even more valuable down a level at the Site level, as I'm thinking about it.

I would second the above, for a widget option to track infrastructure metrics per Tenant, eg overall cables, power, &c. We often get requests asking to calculate eg what percent of a Site is being consumed by a particular Tenant, which that sort of rollup would make much simpler to answer.
